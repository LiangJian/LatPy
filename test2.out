Using numpy, time taken is  21.479623317718506 s.
Using numpy, time taken is  21.534982681274414 s.
Using numpy, time taken is  21.611216068267822 s.
Using numpy, time taken is  21.636613130569458 s.
Using HeAT on cpus, time taken is  7.42155909538269 s.
Using HeAT on cpus, time taken is  7.4291791915893555 s.
Using HeAT on cpus, time taken is  7.475934267044067 s.
Using HeAT on cpus, time taken is  7.484049320220947 s.
Uisng cupy, time taken is  4.637354202270508 s.
Uisng cupy, time taken is  4.645987501144409 s.
Uisng cupy, time taken is  4.667140140533447 s.
Uisng cupy, time taken is  4.669770641326904 s.
Using HeAT on gpus, time taken is  0.3028096652030945 s.
Using HeAT on gpus, time taken is  0.3028931832313538 s.
Using HeAT on gpus, time taken is  0.30307195901870726 s.
Using HeAT on gpus, time taken is  0.4186968183517456 s.
