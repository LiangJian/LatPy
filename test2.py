import heat as ht
import timeit
import numpy as np
import cupy as cp
import time

rank = ht.get_comm().rank
N = 50

a = np.random.rand(10000, 10000)
b = np.random.rand(10000, 10000)
c = np.random.rand(10000, 10000)
# print(type(a[0,0]))
t1 = time.time()
for i in range(N):
  d = c + a * b
t2 = time.time()
print(rank, "Using numpy, time taken is ", t2-t1, "s.")

a = ht.random.rand(10000, 10000, dtype=ht.float64, split=0, device='cpu')
b = ht.random.rand(10000, 10000, dtype=ht.float64, split=0, device='cpu')
c = ht.random.rand(10000, 10000, dtype=ht.float64, split=0, device='cpu')
t1 = time.time()
for i in range(N):
  d = c + a * b
t2 = time.time()
print(rank, "Using HeAT on cpus, time taken is ", t2-t1, "s.")

a = cp.random.rand(10000, 2500)
b = cp.random.rand(10000, 2500)
c = cp.random.rand(10000, 2500)
t1 = time.time()
for i in range(N*100):
  d = c + a * b
t2 = time.time()
print(rank, "Uisng cupy, time taken is ", (t2-t1)*4/100., "s.")

a = ht.random.rand(10000, 10000, dtype=ht.float64, split=0, device='gpu')
b = ht.random.rand(10000, 10000, dtype=ht.float64, split=0, device='gpu')
c = ht.random.rand(10000, 10000, dtype=ht.float64, split=0, device='gpu')
t1 = time.time()
for i in range(N*100):
  d = c + a * b
t2 = time.time()
print(rank, "Using HeAT on gpus, time taken is ", (t2-t1)/100., "s.")


