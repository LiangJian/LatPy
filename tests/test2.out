Using numpy, time taken is  21.72949481010437 s. 
Using HeAT on cpus, time taken is  7.408856153488159 s. 
Uisng cupy, time taken is  4.665657548904419 s. 
Using HeAT on gpus, time taken is  0.4109229111671448 s. 
